hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/utils/MetadataConversionUtils.java

```diff
@@ -105,14 +114,15 @@ public static HoodieArchivedMetaEntry createMetaWrapper(HoodieInstant hoodieInst
     return archivedMetaWrapper;
   }
 
-  public static HoodieArchivedMetaEntry createMetaWrapper(HoodieInstant hoodieInstant,
-                                                          HoodieCommitMetadata hoodieCommitMetadata) {
-    HoodieArchivedMetaEntry archivedMetaWrapper = new HoodieArchivedMetaEntry();
-    archivedMetaWrapper.setCommitTime(hoodieInstant.getTimestamp());
-    archivedMetaWrapper.setActionState(hoodieInstant.getState().name());
-    archivedMetaWrapper.setHoodieCommitMetadata(convertCommitMetadata(hoodieCommitMetadata));
-    archivedMetaWrapper.setActionType(ActionType.commit.name());
-    return archivedMetaWrapper;
+  public static Option<HoodieRequestedReplaceMetadata> getRequestedReplaceMetadata(HoodieTableMetaClient metaClient, HoodieInstant pendingReplaceInstant) throws IOException {
+    final HoodieInstant requestedInstant = HoodieTimeline.getReplaceCommitRequestedInstant(pendingReplaceInstant.getTimestamp());
+
+    Option<byte[]> content = metaClient.getActiveTimeline().getInstantDetails(requestedInstant);
+    if (!content.isPresent() || content.get().length == 0) {
+      LOG.warn("No content found in requested file for instant " + pendingReplaceInstant);
+      return Option.of(new HoodieRequestedReplaceMetadata());
```

Comment:
Why not return Option.empty() and skip archival for this case?

Reply:
Current logic is using `import org.apache.hudi.common.model.HoodieCommitMetadata.fromBytes()` to fetch empty deltacommit (bytes = []), and creating a new instance of metadata which bytes not empty. So I was thinking it may be better to keep same behaviour in ReplaceCommitRequestedInstant.

Reply:
I will try with what you suggested